### ISSUE TITLE
How to use image or combine GVHMR inference?

### ISSUE THREAD

**Issue by psiydown** [Charrrrrlie/Learnable-SMPLify#3]
The result of your paper is impressive, and there are several questions.
How to enter a single custom image inference, How to combine GVHMR or SMPLest-X inference?
Hope to give a code example, thank you.

**Comment by Charrrrrlie** [comment]
Hi @psiydown ! Thank you for your interest in our work!

Please note that our method does not rely on images during inference. As mentioned in the README (tl;dr section), it focuses on the task:

> Given X_{t-s} and X_{t} 3D keypoints, calculate residual SMPL parameters from t-s to t.


To integrate our method with GVHMR or SMPLest-X, we load their prediction results, convert them into world coordinates, and extract the corresponding 3D keypoints for data preparation. We conduct conversion as the current version of our method is trained on 3D, while it also fits for camera-coordinate training.
Here is a simple demonstration. Hope it can help you.

```python

# init net, load SMPL predictions, load ground truth 3D keypoints
smpl_data = xxx
target_joints = xxx
net = xxx

# obtain input joints from init SMPL parameters
smpl_init = net.human_model.layer['neutral'](
        betas=smpl_data['betas'].to(device),
        global_orient=smpl_data['global_orient'].to(device),
        body_pose=smpl_data['body_pose'].to(device)
)

init_verts = smpl_init.vertices
init_pose3d = torch.einsum('bvc,jv->bjc', smpl_init.vertices, net.openpose_regressor)

# <convert keypoints and global rot from camera-coordinates to world-coordinates, using camera parameters> 
input_init_joints = xxxx
input_target_joints = xxxx
global_rot_vec = xxxx.     # (1, 1, 3)
# <end conversion>

input_init_joints, R, T = normalize_kp(input_init_joints, None, kp_index, R=None, T=None)
input_target_joints, _, _ = normalize_kp(input_target_joints, None, kp_index, R=R, T=T)
    
input_joints = torch.stack([input_init_joints, input_target_joints], dim=1)
input_joints = input_joints.permute(0, 3, 1, 2)

# predict
outputs = net.predict(input_joints, global_rot_world_vec, input_body_pose, smpl_data['betas'])

**Comment by Charrrrrlie** [comment]
I’m closing this issue as there are no current updates.
Please feel free to reopen it if you have any further questions. :-)
